Add 2.5D approximation

Huihuiweng opened the pull request:
Define seismogenic width LY in type sem_grid_type and read it from block $GENERAL 
Define beta in type matwrk_elast_type to save dvol*mu*(pi/LY)**2, initiated by subroutine MAT_ELAST_init_CrustalPlane(beta,ngll,dvol,LY,mat,ndof)
Use function MAT_CP_add_f to add the effects of seismogenic boundaries by: if (grid%LY < huge(1d0)) call MAT_CP_add_f(f,d,matwrk%elast,ngll,ndof)

jpampuero commented:
@Huihuiweng please create a branch that I can edit, like @elifo is doing. Let's not merge to the master branch code that is not yet finished, tested and documented.

jpampuero commented:
Some cosmetic suggestions:
In input.f90, replace the name "LY" by the more intuitive "W" (seismogenic Width).
Make LY part of the elastic material type, rather than the grid type.
Replace "MAT_CP_add_f" by "MAT_ELAST_add_25D_f".
Replace "MAT_ELAST_init_CrustalPlane" by "MAT_ELAST_init_25D"

Please create an example in EXAMPLES/25D_rupture/

Huihuiweng commented:
> Some cosmetic suggestions:
> In input.f90, replace the name "LY" by the more intuitive "W" (seismogenic Width).
> Make LY part of the elastic material type, rather than the grid type.
> Replace "MAT_CP_add_f" by "MAT_ELAST_add_25D_f".
> Replace "MAT_ELAST_init_CrustalPlane" by "MAT_ELAST_init_25D"
> 
> Please create an example in EXAMPLES/25D_rupture/

I have replaced several small cosmetic changes and added an example.

>> Make LY part of the elastic material type, rather than the grid type.
I don't quite understand the structure of elastic material type. I don't find where you save the parameters, like rho, cs, cp. Is it a different way from grid type?
If I save W into elastic material type, shall we visit this parameters by call MAT_getProp(W, mat,'W')? If so, it seems not convenient to visit it. And it seems W becomes a property of each element? 
For grid type, we can visit it by grid%xxx easily. Do you know where to put it in the material type (like the way as grid%W) to make it easy to visit?
